ROADMAP.md

```diff
@@ -1,62 +1,191 @@
+- Security
+  - We strive to ensure that libp2p is reliable & secure for users & applications. We seek to minimize vulnerabilities and act quickly to address issues.
+- Stability
+  - We do not break deployed features and maintain cross version compatibility across libp2p releases.
+- Performance
+  - We aim to make libp2p as performant as possible and strive to ensure comparable performance to widely used protocols on the Internet. We avoid regressions and performance degradations in new releases.
+
+This roadmap is complementary to those of [go-libp2p](https://github.com/libp2p/go-libp2p/blob/master/ROADMAP.md), [rust-libp2p](https://github.com/libp2p/rust-libp2p/pull/2997), and [js-libp2p](https://github.com/libp2p/js-libp2p/pull/1412).
+
+It aims to encompass the **stability** and **performance** tenets of the libp2p team.
+Projects outlined here are shared priorities of the different implementations.
+
+### Sections
+This document consists of two sections: [Milestones](#️-milestones) and the [Appendix](#-appendix)
+
+[Milestones](#️-milestones) is our best educated guess (not a hard commitment) around when we plan to ship the key features.
+Where possible projects are broken down into discrete sub-projects e.g. project "A" may contain two sub-projects: A.1 and A.2
+
+A project is signified as "complete" once all of it's sub-projects are shipped.
+
+The [Appendix](#-appendix) section describes a project's high-level motivation, goals, and lists sub-projects.
+
+Each Appendix header is linked to a GitHub Epic. Latest information on progress can be found in the Epics and child issues.
+
+## 🛣️ Milestones
+
+### 2022
+
+#### Early Q4 (October)
+- [A.1 User Configured Interop Tests & Dashboard](#1-user-configured-interop-tests--dashboard)
+
+#### Mid Q4 (November)
+- [A.2 Interop tests for all existing/developing features](#2-interop-tests-for-all-existingdeveloping-features)
+- [A.3 Rock solid libp2p releases](#3-rock-solid-libp2p-releases)
+- [C.1 Benchmarking using nix-builders](#1-benchmarking-using-nix-builders)
+
+#### End of Q4 (December)
+- [A.4 Canonical Interop Tests & Dashboard](#4-canonical-interop-tests--dashboard)
+
+### 2023
+
+#### Early Q1 (January)
+
+- add here
+
+#### Mid Q1 (February)
+
+- add here
+
+#### End of Q1 (March)
+- [C.2 Benchmarking using remote runners](#2-benchmarking-using-remote-runners)
+
+### Up Next
+
+## 📖 Appendix
+
+**Projects are listed in descending priority.**
+
+### A. [Multi-dimensional Testing/Interop Visibility](https://github.com/libp2p/test-plans/issues/53)
+**Why:** libp2p supports a variety of protocols (multiple transports, muxers, & security protocols) across implementations in different languages. Until we actually test them together, we can't guarantee implementation interoperability. We need to ensure and demonstrate that: interoperable features work with each other as expected and we don't introduce degradations that break interoperability in new releases.
```

Comment:
I can go with it, but I wanted to confirm that there is a shared understanding that a "variety of protocols" is our way of talking about the dimensionality of traports, muxers, and security protocols.

Reply:
Down below we use the work "protocols" to mean what I would have expected (e.g., autonat, identity).

What about using the word "modules" (since we describe libp2p as modular) or "options"?

Reply:
> What about using the word "modules" (since we describe libp2p as modular) or "options"?

We don't use the word "module" anywhere else.

I'd suggest renaming this section to include the word "transport", or maybe "libp2p transport". That should make it clear that it encompasses QUIC, WebTransport, WebRTC ect., muxers and security protocols, but not libp2p protocols like autonat, DCUtR etc.

Reply:
Sounds good to me.

I reached for the word "module" from the libp2p being a "[modular network stack](https://libp2p.io/)" "[composed of many modules](https://libp2p.io/implementations/)..."